Tests/Controller/ActionControllerTest.php

```diff
@@ -330,6 +304,36 @@ public function testActionTest()
         $this->assertEquals('{"test":"test"}', $response->getContent());
     }
 
+    public function testActionReadOnly()
+    {
+        $request = $this->prepareRequest('test');
+        $container = self::$container;
+        $container->set("syrup.storage_api", $this->getStorageServiceStubDcaPython(false, 'readOnly'));
+        $container->get('request_stack')->push($request);
+
+        $ctrl = new ActionController();
+        $ctrl->setContainer(self::$container);
+        $ctrl->preExecute($request);
+        $this->expectException(UserException::class);
+        $this->expectExceptionMessage('As a readOnly user you cannot perform any actions.');
+        $ctrl->processAction($request);
```

Comment:
Cakal by som, ze tu sa tiez checkne getStatusCode a ci je to 400ka. Ale chapem, ze to spadne vnutri tak to nie je uplne jednoduche.